pkg/envoy/ads/server.go

```diff
@@ -24,7 +24,7 @@ import (
 const ServerType = "ADS"
 
 // NewADSServer creates a new Aggregated Discovery Service server
-func NewADSServer(meshCatalog catalog.MeshCataloger, enableDebug bool, osmNamespace string, cfg configurator.Configurator, certManager certificate.Manager) *Server {
+func NewADSServer(meshCatalog catalog.MeshCataloger, enableDebug bool, osmNamespace string, cfg configurator.Configurator, certManager certificate.Manager, maxDataPlaneConnections int) *Server {
```

Comment:
no need to pass this argument, it should be dynamically retrieved using `cfg.GetMaxDataPlaneConnections()`